### PR TITLE
pjsip: add tests and darwin compatiblity

### PR DESCRIPTION
**Pull request by bobvanderlinden**
###### Description of changes

Currently pjsip only has tests for checking the whether Python can load the pjsua2 libraries. However, there are no tests that check the executables themselves. Colleagues ran into the `pjsua` executable not working on MacOS.

This PRs goal is to fix Darwin support for pjsip executables and to avoid this in the future.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by bobvanderlinden** [comment]
@NixOS/ofborg was able to run the tests on aarch64-darwin and x86_64-darwin. This includes running `pjsua --version`.

A collegue was able to run this test (and binary) successfully as well.

@LnL7 @olynch Could you have a look for review / merge?

File: pkgs/applications/networking/pjsip/default.nix [hunk 1]
```diff
@@ -98,11 +98,26 @@ stdenv.mkDerivation rec {
         install_name_tool -id $lib "''${change_args[@]}" $lib
       fi
     done
+
+    # Rewrite library references for all executables.
+    find "$out" -executable -type f | while read executable; do
+      install_name_tool "''${change_args[@]}" "$executable"
+    done
```
Comment (wegank) [inline pkgs/applications/networking/pjsip/default.nix:105]: This should only be done on darwin.

Comment (wegank) [reply pkgs/applications/networking/pjsip/default.nix:105]: Oh, you're absolutely right.

Comment (bobvanderlinden) [reply pkgs/applications/networking/pjsip/default.nix:105]: Thanks! :partying_face:

**Pull request merged**